src/cpp/util/byte_buffer.cc

```diff
@@ -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
   return *this;
 }
 
+void ByteBuffer::Swap(ByteBuffer* other) {
+  grpc_byte_buffer* tmp = other->buffer_;
+  other->buffer_ = this->buffer_;
```

Comment:
Yeah, I like an explanation wherever I see "this". There is at least 1 case
where it's needed: template disambiguation. Aside from that, I tend to
think it's confusing. Will send a follow-on PR.

On Thu, Aug 18, 2016 at 9:53 AM Mark D. Roth <notifications@github.com>
wrote:

> In src/cpp/util/byte_buffer.cc
> <https://github.com/grpc/grpc/pull/7773#discussion_r75345611>:
>
> > @@ -95,4 +102,10 @@ ByteBuffer& ByteBuffer::operator=(const ByteBuffer& buf) {
> >    return *this;
> >  }
> >
> > +void ByteBuffer::Swap(ByteBuffer* other) {
> > +  grpc_byte_buffer* tmp = other->buffer_;
> > +  other->buffer_ = this->buffer_;
>
> Yeah, that shouldn't be necessary. I copied that from the internal CL and
> didn't notice it before I merged. Feel free to send me a PR to fix it if it
> bothers you.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/7773/files/2665bdd6d29ec2515946ac629002c04c23a50bfa#r75345611>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIdEkJjQ_u_jjjlxxCjykq9s5ni2gIUZks5qhI4VgaJpZM4Jnln5>
> .
>